angular/projects/admin-nrpti/src/app/records/administrative-penalties/administrative-penalty-add-edit/administrative-penalty-add-edit.component.ts

```diff
@@ -406,11 +406,18 @@ export class AdministrativePenaltyAddEditComponent implements OnInit, OnDestroy
     if (!this.isEditing) {
       this.factoryService.writeRecord(administrativePenalty, 'administrativePenalties', true).subscribe(async res => {
         this.recordUtils.parseResForErrors(res);
+        let _id = null;
+        if (Array.isArray(res[0][0].object)) {
+          _id = res[0][0].object.find(r => r._schemaName === 'AdministrativePenalty')._id;
```

Comment:
Do we have constants available for these?

Reply:
We do for the whole collection of Schema types, but not a constants list of them individually. I can add one if it's preferred to just strings here.